# Check collision of the robot with the environment. collision is 1 if the robot
# is colliding, colbodyid is the id of the object that body collided with
string sessionname
int32 sessionid
uint32 bodyid

# the link of the body to check collision again, if -1 check all links
byte linkid

uint32[] excludedbodyids

# value to specify tolerance of collision checks (only valid if correct option is set), default is 0
float32 tolerance

# optional collision checker options, affects what information is sent back
byte options

byte CO_Distance=1
byte CO_UseTolerance=2
byte CO_Contacts=4
byte CO_RayAnyHit=8

---
byte collision
uint32 collidingbodyid
Contact[] contacts

# minimum distance between the two links if not colliding (only valid if requested)
float32 mindist
